### PR TITLE
LPS-53601 Cleanup objects left in database after test

### PR DESCRIPTION
**Pull request by willnewbury**
@adolfopa - this test generates database data in the following form: we have a repository that is auto-generated and linked to it are two folders, one that is auto-generated as well, and one manually created. In order to clean up after the tests we have to delete both the manually created forlder as well as the repository itself. This would be greatly simplified if the repository deleted all folders attached to it, rather than just the one that it auto-generates itself. This would also make some logical sense, because the repository is created upon the manual folder's creation itself, for the sole purpose of being linked to the folder. 

The current code found in the repository local services makes no effort to eliminate sub-folders upon repo deletion. Would adding this in be appropriate and/or possible?

### CODE REVIEW

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/24213

**Comment by adolfopa** [comment]
Hey @willnewbury,

*TL;DR* To delete a repository and all its resources you should use the `DLApp*Service#deleteAll()` method (or alternatively, delete the repository group, which will delete all the group repositories and
resources).

`Repository*Service` are low level services; their only purpose is to keep track of which external repositories (any repo but the default one) are in use. They had nothing to do with file entries, folders,
etc.

The problem here is that we're using the same name for two different things:
  - `com.liferay.portal.model.Repository`, bookkeeping services to keep track of external repositories.
  - `com.liferay.portal.kernel.repository.Repository`, rich model that exposes CRUD operations over files, folders, etc.

What you really want is to clean up all resources for repositories of the second kind, and for that you need to use the `DLApp*` services.

I know this is extremely confusing, and its a shame we're using the same name for two almost unrelated things (and worse, `Repository*Service` instantiates both kinds of repositories). Eventually, we'll clean up all this mess.

Thanks!

**Comment by willnewbury** [comment]
I see. Thanks for the clarification and the pull request with an updated test!
